test/integration/newrelic-response-handling.tap.js

```diff
@@ -52,6 +52,103 @@ tap.test('New Relic response code handling', (t) => {
+ * the new harvest limit value from server that has been down sampled.  It could result in harvest limits
+ * being 0 if enough restarts occur.
+ */
+tap.test('Connect calls re-generate harvest limits from original config values', (t) => {
+  let agent
+  let serverHarvest
+
+  t.before(() => {
+    serverHarvest = {
+      event_harvest_config: {
+        report_period_ms: 100,
+        harvest_limits: {
+          analytic_event_data: 10,
+          custom_event_data: 10,
+          error_event_data: 1,
+          span_event_data: 10,
+          log_event_data: 10
+        }
+      }
+    }
+    nock.disableNetConnect()
+    nockRequest('preconnect').reply(200, { return_value: TEST_DOMAIN })
+    nockRequest('connect').reply(200, { return_value: { agent_run_id: RUN_ID, ...serverHarvest } })
+    nockRequest('agent_settings', RUN_ID).reply(200, { return_value: [] })
+    nockRequest('metric_data', RUN_ID).reply(409, { return_value: [] })
+    nockRequest('preconnect').reply(200, { return_value: TEST_DOMAIN })
+    nockRequest('connect').reply(200, { return_value: { agent_run_id: RUN_ID, ...serverHarvest } })
+    nockRequest('agent_settings', RUN_ID).reply(200, { return_value: [] })
+    agent = helper.loadMockedAgent({
+      license_key: 'license key here',
+      host: TEST_DOMAIN,
+      application_logging: {
+        enabled: true
+      }
+    })
+  })
+
+  t.teardown(() => {
+    helper.unloadAgent(agent)
+    if (!nock.isDone()) {
+      // eslint-disable-next-line no-console
+      console.error('Cleaning pending mocks: %j', nock.pendingMocks())
+      nock.cleanAll()
+    }
+
+    nock.enableNetConnect()
+  })
+
+  const originalConfig = Object.assign({}, agent.config)
+  agent.config.no_immediate_harvest = true
+  sinon.spy(agent.collector, '_connect')
+
+  /**
+   * This flow starts agent which pre-connects, connects and gets agent settings.
+   * Then we call send metrics and since the metrics collector endpoint is responding
+   * with 409 it will issue a restart and make another pre-connect, connect and agent
+   * settings call.
+   */
+  agent.start((err) => {
+    t.error(err)
+    const config = agent.config
+    t.same(
+      config.event_harvest_config,
+      serverHarvest.event_harvest_config,
+      'config should have been updated from server'
+    )
+    agent.metrics.once('finished metric_data data send.', function onMetricsFinished() {
+      const connectCalls = agent.collector._connect.args
+      t.same(
+        config.event_harvest_config,
+        serverHarvest.event_harvest_config,
+        'config should have been updated from server after reconnect'
+      )
+      t.equal(connectCalls.length, 2, 'should have reconnected once')
+      connectCalls.forEach((call) => {
+        const factsConfig = call[0][0]
+        t.not(
+          factsConfig.event_harvest_config,
+          config.event_harvest_config,
+          'facts harvest config should not be same as new harvest config'
```

Comment:
So this assert passes, but it could also pass for an unrelated reason: `factsConfig.event_harvest_config` doesn't have a `report_period_ms` property but `config.event_harvest_config` does. Can we do a more careful test that actually compares some of the harvest limit values?